src/client/e2e/fixtures.ts

```diff
@@ -0,0 +1,146 @@
+    case "openfin-window-frame?app=CREDIT":
+      return "mainWindow"
+    default:
+      return path as CreditPage
+  }
+}
+
+export const test = base.extend<IPlaywrightFixtures>({
+  browser: async ({}, use) => {
+    try {
+      const runtimeConnection = await chromium.connectOverCDP(RUNTIME_ADDRESS)
+      await use(runtimeConnection)
+    } catch (e) {
+      const browser = await chromium.launch()
+      await use(browser)
+    }
+  },
+  context: async ({ browser }, use) => {
+    try {
+      const contexts = await browser.contexts()
+      if (contexts.length !== 1) {
+        throw Error(
+          `Unexepcted Context(s): Expected 1, Found ${contexts.length}`,
+        )
+      }
+      await use(contexts[0])
+    } catch (e) {
+      const context = await browser.newContext()
+      use(context)
+    }
+  },
+  fxOpenfinPagesRec: async ({ context }, use) => {
+    const contextPages = await context.pages()
+    try {
+      const pages = fxOpenfinUrlPaths.reduce((rec, urlPath) => {
+        const page = contextPages.find(
+          (p) => p.url() === `http://localhost:1917/${urlPath}`,
+        )
+        if (!page) throw Error(`Openfin page at ${urlPath} was not found`)
+        return { ...rec, [urlPathToFxPage(urlPath)]: page }
+      }, {} as Record<FXPage, Page>)
+      use(pages)
+    } catch (e) {
+      const mainWindow =
+        contextPages.length > 0 ? contextPages[0] : await context.newPage()
+      use({
+        mainWindow,
+        "fx-analytics": mainWindow,
+        "fx-blotter": mainWindow,
+        "fx-tiles": mainWindow,
+      })
+    }
+  },
+  creditOpenfinPagesRec: async ({ context }, use) => {
+    const contextPages = await context.pages()
+    try {
+      const pages = creditOpenfinUrlPaths.reduce((rec, urlPath) => {
+        const page = contextPages.find(
+          (p) => p.url() === `http://localhost:1917/${urlPath}`,
+        )
+        if (!page) throw Error(`Openfin page at ${urlPath} was not found`)
+        return { ...rec, [urlPathToCreditPage(urlPath)]: page }
+      }, {} as Record<CreditPage, Page>)
+      use(pages)
+    } catch (e) {
+      const mainWindow =
+        contextPages.length > 0 ? contextPages[0] : await context.newPage()
+      use({
+        mainWindow,
+        "credit-blotter": mainWindow,
+        "credit-new-rfq": mainWindow,
+        "credit-rfqs": mainWindow,
+      })
+    }
+  },
+  openfinNotification: async ({}, use) => {
+    const runtimeConnection = await chromium.connectOverCDP(
+      "http://localhost:9090",
+    )
+
+    const contexts = runtimeConnection.contexts()
+    if (contexts.length !== 1) {
+      throw Error(
+        `Unexepcted Context(s) for Notification: Expected 1, Found ${contexts.length}`,
+      )
+    }
+
+    const pages = await contexts[0].pages()
+
+    const notificationPage = pages.find(
+      (page) =>
+        page.url() ===
+        "https://cdn.openfin.co/services/openfin/notifications/0.12.10/provider.html",
```

Comment:
as we saw today, this is dependent on the nature of the manifest in question .. can we extract into env do you think?
we could do the same with the localhost:9090 OpenFin debug address above too, perhaps